### DISCUSSION TITLE
Hierarchy/priority order of rewrites

### DISCUSSION THREAD

**Discussion by darbymanning** [vercel/community#432] [Help]
I'm attempting to perform some rewrites on a SvelteKit app using a `vercel.json` file [as per the docs](https://vercel.com/docs/cli#project-configuration/rewrites). This is working for me at a basic level, but I'm getting a little stuck as I try to advance my usage. What I'm hoping to achieve is:

- By default, have _all URLs_ rewrite to https://example.com (✅ working)
- Add an exclusion for the routes which should route to the local SvelteKit App:
  - `/collections/:collection/products/:handle` (✅ working)
  - `/products/:handle` (❌ not working)

As it's so specific/nested I've been able to get this working well for the collections product route, but not for the `/products/:handle` route (it just routes to https://example.com/products/:handle instead of the SvelteKit app).

Here's my `vercel.json`:

```json
{
   "rewrites": [
      {
         "source": "/collections/:collection/products/:handle*",
         "destination": "/collections/:collection/products/:handle*"
      },
      {
         "source": "/products/:handle*",
         "destination": "/products/:handle*"
      },
      {
         "source": "/",
         "destination": "https://www.example.com"
      },
      {
         "source": "*",
         "destination": "https://www.example.com/*"
      }
   ]
}
```

How I can get `products/:handle` to work? Perhaps by telling it it's "higher priority" than the wildcard that is currently matching, or something like that?

**Comment by samcx** [comment]
Is there anything different between these two routes?

```
{
  "source": "/collections/:collection/products/:handle*",
  "destination": "/collections/:collection/products/:handle*"
},
{
  "source": "/products/:handle*",
  "destination": "/products/:handle*"
},
```

Maybe you could do some advanced regex instead in the catch-all. →

```
{
  "source": "/:path((?!product/).*)",
  "destination": "https://example.com/:path*"
}
```
